IAM permission issue with spanner

The default credentials to GCP has owner, editor and cloud spanner admin role for the project . However , the code keeps showing IAM permission issue

`Caused by: io.grpc.StatusRuntimeException: PERMISSION_DENIED: Resource projects/gcp-ftd-nonprod-db/instances/testspanner is missing IAM permission: spanner.databases.list.`

When a service account (with json key) is used instead  , it  has the same permission issue with different message

`Caused by: java.util.concurrent.ExecutionException: io.grpc.StatusRuntimeException: PERMISSION_DENIED: Request had insufficient authentication scopes.`

Here is what application.properties have

```
spring.cloud.gcp.spanner.project-id=gcp-ftd-nonprod-db
spring.cloud.gcp.spanner.instance-id=testspanner
spring.cloud.gcp.spanner.database=service1db
spring.cloud.gcp.spanner.credentials.location=file:/Users/vsekhri/apps/gcp/gcp-nonprod-test-spanner.json
```

Same  credentials work when used on the console or with a jdbc driver 
```
<dependency>
            <groupId>nl.topicus</groupId>
            <artifactId>spanner-jdbc</artifactId>
            <version>0.19</version>
            <classifier>shaded</classifier>
        </dependency>
```

Here are the permission for the account used 

![image](https://user-images.githubusercontent.com/6306308/40556282-bd245068-6011-11e8-8bd5-ed5c44a65274.png)

It is the standard code for the sample app failing at tableExist call
https://github.com/spring-cloud/spring-cloud-gcp/blob/master/spring-cloud-gcp-samples/spring-cloud-gcp-data-spanner-sample/src/main/java/com/example/SpannerRepositoryExample.java#L108


```

Caused by: com.google.cloud.spanner.SpannerException: PERMISSION_DENIED: io.grpc.StatusRuntimeException: PERMISSION_DENIED: Request had insufficient authentication scopes.
	at com.google.cloud.spanner.SpannerExceptionFactory.newSpannerExceptionPreformatted(SpannerExceptionFactory.java:119) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.SpannerExceptionFactory.newSpannerException(SpannerExceptionFactory.java:43) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.SpannerExceptionFactory.newSpannerException(SpannerExceptionFactory.java:80) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.spi.v1.GrpcSpannerRpc.get(GrpcSpannerRpc.java:456) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.spi.v1.GrpcSpannerRpc.listDatabases(GrpcSpannerRpc.java:274) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.SpannerImpl$DatabaseAdminClientImpl$6.getNextPage(SpannerImpl.java:559) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.SpannerImpl$PageFetcher$1.call(SpannerImpl.java:398) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.SpannerImpl$PageFetcher$1.call(SpannerImpl.java:395) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.SpannerImpl.runWithRetries(SpannerImpl.java:227) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.SpannerImpl$PageFetcher.getNextPage(SpannerImpl.java:394) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at com.google.cloud.spanner.SpannerImpl$DatabaseAdminClientImpl.listDatabases(SpannerImpl.java:570) ~[google-cloud-spanner-0.47.0-beta.jar:0.47.0-beta]
	at org.springframework.cloud.gcp.data.spanner.core.admin.SpannerDatabaseAdminTemplate.databaseExists(SpannerDatabaseAdminTemplate.java:127) ~[classes/:na]
	at org.springframework.cloud.gcp.data.spanner.core.admin.SpannerDatabaseAdminTemplate.tableExists(SpannerDatabaseAdminTemplate.java:193) ~[classes/:na]
	at com.example.SpannerRepositoryExample.createTablesIfNotExists(SpannerRepositoryExample.java:108) [classes/:na]
	at com.example.SpannerRepositoryExample.lambda$commandLineRunner$1(SpannerRepositoryExample.java:59) [classes/:na]

```



I will remove the .m2 directory and try again just to eliminate the possibility of wrong dependencies

--- a/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerProperties.java
+++ b/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/spanner/GcpSpannerProperties.java
@@ -31,7 +31,8 @@ public class GcpSpannerProperties implements CredentialsSupplier {
 
 	/** Overrides the GCP OAuth2 credentials specified in the Core module. */
 	@NestedConfigurationProperty
-	private final Credentials credentials = new Credentials(GcpScope.SPANNER.getUrl());
+	private final Credentials credentials = new Credentials(
+			GcpScope.SPANNER_DATA.getUrl(), GcpScope.SPANNER_ADMIN.getUrl());
 
 	private String projectId;
 

--- a/spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/GcpScope.java
+++ b/spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/GcpScope.java
@@ -23,7 +23,8 @@ package org.springframework.cloud.gcp.core;
  */
 public enum GcpScope {
 	PUBSUB("https://www.googleapis.com/auth/pubsub"),
-	SPANNER("https://www.googleapis.com/auth/spanner.data"),
+	SPANNER_ADMIN("https://www.googleapis.com/auth/spanner.admin"),
+	SPANNER_DATA("https://www.googleapis.com/auth/spanner.data"),
 	SQLADMIN("https://www.googleapis.com/auth/sqlservice.admin"),
 	STORAGE_READ_ONLY("https://www.googleapis.com/auth/devstorage.read_only"),
 	STORAGE_READ_WRITE("https://www.googleapis.com/auth/devstorage.read_write"),

